manifests/components/kibana.jsonnet

```diff
@@ -38,18 +39,69 @@ local strip_trailing_slash(s) = (
     },
   },
 
+  // List of Kibana plug-ins to install
+  plugins:: {
+    /* Example:
+    logtrail: {
+      version: "v0.1.31",
+      url: "https://github.com/sivasamyk/logtrail/releases/download/v0.1.31/logtrail-6.7.1-0.1.31.zip",
+    },
+    */
+  },
+
   es: error "elasticsearch is required",
 
   serviceAccount: kube.ServiceAccount($.p + "kibana") + $.metadata {
   },
 
+  pvc: kube.PersistentVolumeClaim($.p + "kibana-plugins") + $.metadata {
+    storage: "2Gi",
+  },
+
   deploy: kube.Deployment($.p + "kibana") + $.metadata {
     spec+: {
       template+: {
         spec+: {
           securityContext: {
             fsGroup: 1001,
           },
+          volumes_+: {
+            plugins: kube.PersistentVolumeClaimVolume($.pvc),
+          },
+          initContainers_+: {
+            kibana_plugins_install: kube.Container("kibana-plugins-install") {
+              image: KIBANA_IMAGE,
+              securityContext: {
+                allowPrivilegeEscalation: false,
+              },
+              env_+: {
+                local plugins = [{name: name} + $.plugins[name] for name in std.objectFields($.plugins)],
+                KIBANA_PLUGINS: std.join('\n', ["%(name)s,%(version)s,%(url)s" % plugin for plugin in plugins]),
+              },
+              command: [
+                "/bin/bash",
+                "-c",
+                |||
+                  set -e
+                  set -x
+                  exec &> >(tee /opt/bitnami/kibana/plugins/log)
```

Comment:
Is it expected that the logs are persisted here, rather than (only) in the k8s initcontainer log?
Unless there is a need/expectation that they will be here, I suggest we just log to stdout/stderr, as usual.

Reply:
Then once the init container is done, how do you access its logs?

Reply:
> how do you access its logs?

`kubectl logs kibana-xxx -c kibana-plugins-install`